### PR TITLE
EOS-13996:ADDB:Profiling for all I/O operations (utils repo)

### PR DESCRIPTION
**Pull request by pratyush-seagate**
#  EOS-13996:ADDB:Profiling for all I/O operations (utils repo)

## Checklist
- [x] **Compilation:** _This patch does not break compilation_
- [x] **Merge conflicts:** _This patch has been squashed and re-based, it can be merged using fast-forward merge_
- [] **Code review:** _All discussions have been resolved_
- [x] **Sanity Testing:** _All Unit tests are passing and able to do mount and io operations works from NFS client_
- [x] **Documentation:** _This patch and merge request have up to date description_
- [x] **Unit Testing and debugging:** _Both single and multi node changes are unit tested_

## UT

[root@ssc-vm-c-0040 /]# ./cortx-posix/scripts/test.sh
 --
 --
Configuration Completed
Clean indexes prepared
NSAL Unit tests
NS Tests
Test results are logged to /var/log/cortx/test/ut/ut_nsal.log
Total tests  = 5
Tests passed = 5
Tests failed = 0

Iterator test
Test results are logged to /var/log/cortx/test/ut/ut_nsal.log
Total tests  = 2
Tests passed = 2
Tests failed = 0

KVTree test
Test results are logged to /var/log/cortx/test/ut/ut_nsal.log
Total tests  = 17
Tests passed = 17
Tests failed = 0

Global KVS Tests
Test results are logged to /var/log/cortx/test/ut/ut_nsal.log
Total tests  = 4
Tests passed = 4
Tests failed = 0

CORTXFS Unit tests
Endpoint ops Tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.logs
Total tests  = 4
Tests passed = 4
Tests failed = 0

FS Tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 3
Tests passed = 3
Tests failed = 0

Directory tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 14
Tests passed = 14
Tests failed = 0

File creation tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 4
Tests passed = 4
Tests failed = 0

Link Tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 6
Tests passed = 6
Tests failed = 0

Rename tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 3
Tests passed = 3
Tests failed = 0

Attribute Tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 5
Tests passed = 5
Tests failed = 0

Xattr file Tests
Test results are logged to /var/log/cortx/test/ut/xattr_file_ops.log
Total tests  = 9
Tests passed = 9
Tests failed = 0

Xattr dir Tests
Test results are logged to /var/log/cortx/test/ut/xattr_dir_ops.log
Total tests  = 9
Tests passed = 9
Tests failed = 0

IO tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 7
Tests passed = 7
Tests failed = 0

DSAL Unit tests
Dsal basic test
Test results are logged to /var/log/cortx/test/ut/ut_dsal.log
Total tests  = 11
Tests passed = 11
Tests failed = 0

Dsal IO test
Test results are logged to /var/log/cortx/test/ut/ut_dsal.log
Total tests  = 2
Tests passed = 2
Tests failed = 0

Dsal space stats test
Test results are logged to /var/log/cortx/test/ut/ut_dsal.log
Total tests  = 2
Tests passed = 2
Tests failed = 0

## Commit Message
commit de4a233b42d996e91eb072eb9e9b82d0abbf3b55
Author: pratyush-seagate <pratyush.k.khan@seagate.com>
Date:   Tue Oct 27 16:10:25 2020 -0600

            EOS-13996:ADDB:Profiling for all I/O operations (utils repo)

            List of modified files:
                modified:   c-utils/src/addb/plugin.c
                modified:   c-utils/src/include/perf/perf-counters.h
                modified:   c-utils/src/test/perf/perf-manual-test.c
            Change description:
                Support for a tag to identify submodules based on function tags
                Write attr fixes
                Unit test:
                    histogram generation

### CODE REVIEW

File: c-utils/src/include/perf/perf-counters.h [hunk 1]
```diff
@@ -132,9 +132,25 @@ struct perfc_callstack {
...
+	PST_INIT,
+	PST_M0KVS,
+	PST_M0_KEY_ITER,
+	PST_END
+};
+
 enum perfc_function_tags {
 	PFT_START,
-
+	// PST_FSAL
```
Comment (vimalraghwani20) [inline c-utils/src/include/perf/perf-counters.h:152]: remove this commented code if not useful in future

**Review by vimalraghwani20 — APPROVED** [review]

File: c-utils/src/include/perf/perf-counters.h (hunk 1 above)
Comment (pratyush-seagate) [reply c-utils/src/include/perf/perf-counters.h:152]: this is not a commented out code. It comments which sub-module tag is associated with which group of function tags.

Comment (AtitaShirwaikar) [reply c-utils/src/include/perf/perf-counters.h:152]: You could name it as // PST_FSAL tags

**Review by AtitaShirwaikar — APPROVED** [review]
+1

**Review by shahidSTX — APPROVED** [review]
LGTM

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**